druid/src/widget/tabs.rs

```diff
@@ -0,0 +1,1006 @@
+        self
+    }
+
+    /// Put the tab bar at the corresponding end of the cross axis.
+    /// Defaults to Start. Note that Middle has the same effect as Start.
+    pub fn with_cross_axis_alignment(mut self, cross: CrossAxisAlignment) -> Self {
+        self.cross = cross;
+        self
+    }
+
+    /// Use the provided transition when tabs change
+    pub fn with_transition(mut self, transition: TabsTransition) -> Self {
+        self.transition = transition;
+        self
+    }
+
+    /// Available when the policy implements AddTab - e.g StaticTabs.
+    /// Return this Tabs widget with the named tab added.
+    pub fn with_tab(
+        mut self,
+        name: impl Into<LabelText<TP::Input>>,
+        child: impl Widget<TP::Input> + 'static,
+    ) -> Tabs<TP>
+    where
+        TP: AddTab,
+    {
+        self.add_tab(name, child);
+        self
+    }
+
+    /// Available when the policy implements AddTab - e.g StaticTabs.
+    /// Return this Tabs widget with the named tab added.
+    pub fn add_tab(
+        &mut self,
+        name: impl Into<LabelText<TP::Input>>,
+        child: impl Widget<TP::Input> + 'static,
+    ) where
+        TP: AddTab,
+    {
+        if let TabsContent::Building { tabs } = &mut self.content {
+            TP::add_tab(tabs, name, child)
+        } else {
+            log::warn!("Can't add static tabs to a running or complete tabs instance!")
+        }
+    }
+
+    fn make_scope(&self, tabs_from_data: TP) -> WidgetPod<TP::Input, TabsScope<TP>> {
+        let (tabs_bar, tabs_body) = (
+            (TabBar::new(self.axis, self.cross), 0.0),
+            (
+                TabsBody::new(self.axis, self.transition)
+                    .padding(5.)
+                    .border(theme::BORDER_DARK, 0.5),
+                1.0,
+            ),
+        );
+        let mut layout: Flex<TabsState<TP>> = Flex::for_axis(self.axis.cross());
+
+        if let CrossAxisAlignment::End = self.cross {
+            layout.add_flex_child(tabs_body.0, tabs_body.1);
+            layout.add_flex_child(tabs_bar.0, tabs_bar.1);
+        } else {
+            layout.add_flex_child(tabs_bar.0, tabs_bar.1);
+            layout.add_flex_child(tabs_body.0, tabs_body.1);
+        };
+
+        WidgetPod::new(Scope::new(
+            TabsScopePolicy::new(tabs_from_data, 0),
+            Box::new(layout),
+        ))
+    }
+}
+
+impl<TP: TabsPolicy> Widget<TP::Input> for Tabs<TP> {
+    fn event(&mut self, ctx: &mut EventCtx, event: &Event, data: &mut TP::Input, env: &Env) {
+        if let TabsContent::Running { scope } = &mut self.content {
+            scope.event(ctx, event, data, env);
+        }
+    }
+
+    fn lifecycle(
+        &mut self,
+        ctx: &mut LifeCycleCtx,
+        event: &LifeCycle,
+        data: &TP::Input,
+        env: &Env,
+    ) {
+        if let LifeCycle::WidgetAdded = event {
+            let mut temp = TabsContent::Swapping;
+            std::mem::swap(&mut self.content, &mut temp);
```

Comment:
nit, but I think the idiomatic way to do this would be to use `std::mem::replace`, something like,

```rust
let content  = std::mem::replace(&mut self.content, TabsContent::Swapping);
```

Reply:
Yep, I think its another case where I didn't know about replace when I wrote this!